docusaurus/docs/React/guides/theming/message-ui.mdx

```diff
@@ -3,273 +3,423 @@ id: message_ui
 title: Message UI
 ---
 
-import CustomMessage from '../../assets/CustomMessage.png';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
 
-## How-to Guide for Building a Custom Message
+Message UI is one of the main building blocks of the chat application. Designing proper message UI is no easy feat and that's why our SDK comes with a pre-built component ([`MessageSimple`](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Message/MessageSimple.tsx)) which is packed with functionality and is easy to customize through [our CSS variables](../../theming/component-variables.mdx) or through component replacement ([Component Context](../../components/contexts/component-context.mdx#message)).
 
-Using pre-built UI components from the library, the following guide walks you through the process of building a custom
-[Message UI](../../components/message-components/message-ui.mdx) component that looks great and comes packed with functionality. This
-component must be added to either the `Channel` or `MessageList` component via the `Message` prop.
+In this guide we'll build a simplified custom message UI component combining pre-built and other completely custom components.
 
-All the message UI components in this example have access to the `MessageContext`, and therefore can call the `useMessageContext`
-custom hook. The return value from this hook contains all the values and functions needed to build a Message UI component.
+Message UI and all of its children can access [`MessageContext`](../../components/contexts/message-context.mdx) in which it's wrapped, and therefore can call the `useMessageContext` hook accessing all of the message related information and functions alike.
 
-### UI Component Overview
+## Custom Message UI
```

Comment:
I think we can drop this header to keep the article structure flat.

Reply:
I was pointing out something similar in your guide, haha, will definitely remove, thanks!